doc/python/asrc_utils.py

```diff
@@ -387,9 +406,39 @@ def run_c_model(self, ipFiles, opRate, blocksize, fDev):
                 os.remove("./input.dat")
                 os.remove("./output.dat")
 
+            # ASRC XSIM Model
+            # ---------------
+            if ipFile[2] in self.srcRates and self.opRate in self.srcRates and not self.skip_xsim:
+                xsimFiles = self.run_xsim(ipFiles, opRate, fDev, simLog)
+                opFiles  = np.append(opFiles, xsimFiles, axis=0)
+
 
         return opFiles, simLog
 
+    def run_xsim(self, ipFiles, opRate, fDev, simLog):
+        print("In run_xsim!!!\n\n")
+
+        opFiles=np.empty((0,6), str)
```

Comment:
len(allrates)?

Reply:
This is creating an array to store 6 strings. We later do, opFiles = np.append(opFiles, np.array([[ch0, ch1, opRate, str(fDev), output, "x-asrc"]]), axis=0)

Looking at this, we don't need a numpy array to store strings, a list would do but I haven't changed the existing code, only added support for running the DUT as part of this test.